census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""This code implements a Feed forward neural network using TF low level APIs.
+   It implements a binary classifier for Census Income Dataset.
+"""
+
+
+import argparse
+import json
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+from tensorflow.contrib.layers.python.ops import sparse_feature_cross_op
+from tensorflow.contrib.layers.python.ops import bucketization_op
+
+from StringIO import StringIO
+
+import model
+import os
+
+#See tutorial on wide and deep https://www.tensorflow.org/tutorials/wide_and_deep/
+#https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/feature_column.py
+
+# csv columns in the input file
+CSV_COLUMNS = ('age', 'workclass', 'fnlwgt', 'education', 'education_num',
+               'marital_status', 'occupation', 'relationship', 'race', 'gender',
+               'capital_gain', 'capital_loss', 'hours_per_week', 'native_country',
+               'income_bracket')
+
+CATEGORICAL_COLS = ('gender', 'race', 'education', 'marital_status',
+                    'relationship', 'workclass','occupation', 'native_country')
+
+CONTINUOUS_COLS = ('age', 'education_num', 'capital_gain', 'capital_loss', 'hours_per_week')
+
+LABEL_COL = 'income_bracket'
+
+
+#
+# Graph creation section
+# The three functions below create everything for the graph.
+#
+def loss(model, labels):
+  """Compute cross entropy loss function."""
+  cross_entropy = tf.reduce_mean(
+      tf.nn.softmax_cross_entropy_with_logits(logits=model, labels=labels))
+  return cross_entropy
+
+def optimizer(loss, global_step):
```

Comment:
Sure, but it has to be invoked from two places 1) single 2) dist. I could remove these functions and create one function which creates graph (is what you are saying). I will still have to create that function because it will be invoked from two different places (1) and (2)